src/vm/jitinterface.cpp

```diff
@@ -6631,6 +6632,12 @@ const char* CEEInfo::getMethodNameFromMetadata(CORINFO_METHOD_HANDLE ftnHnd, con
         {
             MethodTable* pMT = ftn->GetMethodTable();
             classResult = pMT->GetFullyQualifiedNameInfo(&namespaceResult);
```

Comment:
I would streamline the implementation to avoid redundant GetMDImport lookups, and also make it throw on errors. We should not ever get bad metadata here, but if we do ... it is better to not ignore it silently.

```
if (!IsNilToken(token))
{
     MethodTable* pMT = ftn->GetMethodTable();
     IMDInternalImport* pMDImport = pMT->GetMDImport();

     IfFailThrow(pMDImport->GetNameOfMethodDef(token, &result));
     IfFailThrow(pMDImport->GetNameOfTypeDef(pMT->GetCl(), &classResult, &namespaceResult));

     if (pMT->GetClass()->IsNested())
     {
         IfFailThrow(pMDImport->GetNameOfTypeDef(pMT->GetEnclosingCl(), &enclosingResult, &namespaceResult));
     }
}
```

Reply:
Thank you so much, will change.